homeassistant/components/mqtt/light/schema_basic.py

```diff
@@ -525,29 +529,45 @@ def rgbw_received(msg: ReceiveMessage) -> None:
             if rgbw is None:
                 return
             self._attr_rgbw_color = cast(tuple[int, int, int, int], rgbw)
-            get_mqtt_data(self.hass).state_write_requests.write_state_request(self)
 
         add_topic(CONF_RGBW_STATE_TOPIC, rgbw_received)
 
         @callback
         @log_messages(self.hass, self.entity_id)
+        @write_state_on_attr_change(
+            self, {"_attr_brightness", "_attr_color_mode", "_attr_rgbww_color"}
+        )
         def rgbww_received(msg: ReceiveMessage) -> None:
             """Handle new MQTT messages for RGBWW."""
+
+            def _converter(
+                r: int, g: int, b: int, cw: int, ww: int
+            ) -> tuple[int, int, int]:
+                min_kelvin = color_util.color_temperature_mired_to_kelvin(
+                    self.min_mireds
+                )
+                max_kelvin = color_util.color_temperature_mired_to_kelvin(
+                    self.min_mireds
+                )
+                return color_util.color_rgbww_to_rgb(
+                    r, g, b, cw, ww, min_kelvin, max_kelvin
+                )
+
             rgbww = _rgbx_received(
                 msg,
                 CONF_RGBWW_VALUE_TEMPLATE,
                 ColorMode.RGBWW,
-                color_util.color_rgbww_to_rgb,
+                _converter,
             )
```

Comment:
This is bug, to be fixed in a separate PR.
Function `color_util.color_rgbww_to_rgb` misses 2 positional attributes: `min_kelvin` and `max_kelvin`.

Without the fix the CI tests will not pass.

Reply:
Moved fix and additional tests to https://github.com/home-assistant/core/pull/100707. To keep this clean.